app/controllers/api/v1/credentials_manager.rb

```diff
@@ -0,0 +1,6 @@
+class Api::V1::CredentialsManager < ApplicationController
```

Comment:
Recommendation for Refactoring: I like the function you made here, but I don't think it needs to be in an `ApplicationController`. Just make a plain old Ruby class in `lib` with this function and use it wherever it is needed.